### ISSUE TITLE
[java] JUnitTestsShouldIncludeAssert - false positives with lambdas and static methods

### ISSUE THREAD

**Issue by hgschmie** [pmd/pmd#2147]
**Affects PMD Version:** 

6.20.0

**Rule:**

[JUnitTestsShouldIncludeAssert](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#junittestsshouldincludeassert)

**Description:**

asserts in static method calls and in lambdas are ignored, triggering false positives

**Code Sample demonstrating the issue:**

```java
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;
import com.fasterxml.jackson.databind.JsonNode;


@Test
public void testDemo() {
  JsonNode jsonNode = assertSuccessResponse();
  final int totalFieldsCount = jsonNode.get("count").asInt();
  jsonNode.get("data").forEach(item -> assertTrue(item.size() <= totalFieldsCount));
}

 public static JsonNode assertSuccessResponse() {
    JsonNode jsonNode = doStuffToBuildTheJsonNode();
    assertTrue(jsonNode.hasNonNull("count"));
    assertTrue(jsonNode.hasNonNull("data"));
    assertTrue(jsonNode.get("count").asInt() > 0);
    assertEquals(jsonNode.get("count").asInt(), jsonNode.get("data").size());
    return jsonNode;
}
```

(In the actual failing code, the assertSuccessResponse method is in a different class as a static helper method).

```
[INFO] PMD Failure: TheTest:98 Rule:JUnitTestsShouldIncludeAssert Priority:3 JUnit tests should include assert() or fail().
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

Maven, OpenJDK from AdoptOpenJDK 1.8

**Comment by adangel** [comment]
Fixed via #2899 for PMD 7.

**Comment by otorasso-nex** [comment]
This is still failing. The example code is passing just because the static method name starts with 'assert'

**Comment by adangel** [comment]
@otorasso-nex Thanks for the info. As this issue is closed, would you mind to create a new issue with the proper code, that is still failing? Thanks!
